core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -308,7 +308,7 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
   val brokerIdGenerationEnable: Boolean = getBoolean(ServerConfigs.BROKER_ID_GENERATION_ENABLE_CONFIG)
   val maxReservedBrokerId: Int = getInt(ServerConfigs.RESERVED_BROKER_MAX_ID_CONFIG)
   var brokerId: Int = getInt(ServerConfigs.BROKER_ID_CONFIG)
-  val nodeId: Int = getInt(KRaftConfigs.NODE_ID_CONFIG)
+  var nodeId: Int = getInt(KRaftConfigs.NODE_ID_CONFIG)
```

Comment:
Could you please add access limit to node id and broker id? for example:
```scala
  private[server] var _nodeIf: Int = getInt(KRaftConfigs.NODE_ID_CONFIG)
  def nodeId: Int = _nodeIf
```

That can't protect us 100% from bugs, but 50% at least I think :smile: